### PR TITLE
Cache incoming data

### PR DESCRIPTION
**Pull request by emanueldima**
This implements the HTTP cache on incoming resources, using Apache HTTP library. 
**We have a new config entry that sets the maximum number of entries in the cache**
Right now the cache is stored in memory.

### CODE REVIEW

File: backend/src/main/java/eu/clarin/switchboard/core/LinkMetadata.java [hunk 1]
```diff
@@ -28,36 +49,128 @@ public String getFilename() {
...
+                    LOGGER.debug("The response came from an upstream server");
+                    break;
+                case VALIDATED:
+                    LOGGER.debug("The response was generated from the cache after validating the entry with the origin server");
+                    break;
+            }
+
+            Header header = response.getFirstHeader("Content-Disposition");
+            if (header != null) {
+                System.out.println("httpget header: " + header.toString());
```
Comment (andmor-) [inline backend/src/main/java/eu/clarin/switchboard/core/LinkMetadata.java:94]: Was this println forgotten here?

Comment (emanueldima) [reply backend/src/main/java/eu/clarin/switchboard/core/LinkMetadata.java:94]: Yes it was! :)

**Comment by andmor-** [comment]
Merging for test. One question:
Is it difficult to store this cache in disk? I assume this will try to cache the files fetched by the Switchboard which can have up to 20MB each. We could run into memory scaling issues. Also cache persistence could be interesting (something to discuss later).

**Comment by emanueldima** [comment]
Should take a few hours to implement that, there is no support out of the box for it. For now we can control the number of entries in the cache.

**Comment by andmor-** [comment]
OK. Sounds good, thanks.

**Pull request merged**